hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -0,0 +1,180 @@
+package openstack
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/support/upsert"
+
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OpenStack struct {
+	capiProviderImage string
+}
+
+func New(capiProviderImage string) *OpenStack {
+	return &OpenStack{
+		capiProviderImage: capiProviderImage,
+	}
+}
+
+func (a OpenStack) ReconcileCAPIInfraCR(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster,
+	controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
+	return nil, nil
+}
+
+func (a OpenStack) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
+	// image := a.capiProviderImage
```

Comment:
This will be uncommented and adjusted later when we enable the CAPI provider cluster. Please ignore the comments.